pallets/storage-proofs/src/lib.rs

```diff
@@ -0,0 +1,483 @@
+        ///
+        /// Finally, it cleans up:
+        /// - The `BlockToChallenges` StorageMap, removing entries older than `ChallengeHistoryLength`.
+        /// - The `BlockToChallengedSps` StorageMap, removing entries for the current block number.
+        #[pallet::call_index(2)]
+        #[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
+        pub fn new_challenges_round(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+            // Inherents are not signed transactions.
+            ensure_none(origin)?;
+
+            // TODO: Handle result of verification.
+            Self::do_new_challenges_round()?;
+
+            // TODO: Emit events.
+
+            // Return a successful DispatchResultWithPostInfo
+            Ok(().into())
+        }
+    }
+
+    // This pallet provides an inherent, as such it implements ProvideInherent trait
+    // https://paritytech.github.io/substrate/master/frame_support/inherent/trait.ProvideInherent.html
+    #[pallet::inherent]
+    impl<T: Config> ProvideInherent for Pallet<T> {
+        type Call = Call<T>;
+        // TODO: Specify this type.
+        type Error = InherentError;
+        // TODO: Specify this type.
+        const INHERENT_IDENTIFIER: InherentIdentifier = INHERENT_IDENTIFIER;
+
+        // This method is used to decide whether this inherent is requiered for the block to be accepted
+        fn is_inherent_required(data: &InherentData) -> Result<Option<Self::Error>, Self::Error> {
+            // TODO: Implement this method
+            unimplemented!()
+        }
+
+        fn create_inherent(data: &InherentData) -> Option<Self::Call> {
+            // create and return the extrinsic call if the data could be read and decoded
+            // TODO: Implement this method
+            unimplemented!()
+        }
+        // Determine if a call is an inherent extrinsic
+        fn is_inherent(call: &Self::Call) -> bool {
+            // TODO: Implement this method
+            unimplemented!()
+        }
+    }
+}
+
+#[derive(Encode)]
+#[cfg_attr(feature = "std", derive(Debug, Decode))]
+pub enum InherentError {
+    Other(RuntimeString),
+}
+
+impl IsFatalError for InherentError {
+    fn is_fatal_error(&self) -> bool {
+        match *self {
+            InherentError::Other(_) => true,
+        }
+    }
+}
+
+impl InherentError {
+    /// Try to create an instance ouf of the given identifier and data.
+    #[cfg(feature = "std")]
+    pub fn try_from(id: &InherentIdentifier, data: &[u8]) -> Option<Self> {
+        if id == &INHERENT_IDENTIFIER {
+            <InherentError as codec::Decode>::decode(&mut &data[..]).ok()
+        } else {
+            None
+        }
+    }
+}
+
+// TODO: Move this to Storage Providers pallet.
+/// A trait to lookup registered Storage Providers.
+///
+/// It is abstracted over the `AccountId` type, `StorageProvider` type, total number of users
+/// and Balance type.
+pub trait StorageProvidersInterface {
```

Comment:
This trait should probably be placed in the storage provider pallet itself since both this pallet and the file system pallet depend on it.